data/mods/ssb/abilities.ts

```diff
@@ -25,6 +25,32 @@ export const BattleAbilities: {[k: string]: ModdedAbilityData} = {
 			}
 		},
 	},
+	// Darth
+	guardianangel: {
+		desc: "Regenerator effect and changes type on switch in to be super effective against foe.", // long description
+		shortDesc: "Regenerator + changes type for super effectiveness against opponent.",
+		name: "Guardian Angel",
+		onSwitchOut(pokemon) {
+			pokemon.heal(pokemon.baseMaxhp / 3);
+		},
+		onStart(pokemon) {
+			let resistances = [];
+			Object.keys(Dex.data.TypeChart).forEach(function(type) {
+				if (Dex.getEffectiveness(type, pokemon.side.foe) < 1) return resistances.push(type);
+				return null;
+			}
+			let randType = resistances[Math.floor(Math.random() * resistances.length)];
+			let randTypeTwo = resistances[Math.floor(Math.random() * resistances.length)];
+			while (randType === randTypeTwo) {
```

Comment:
infinite loop warning

Reply:
Unless you're incredibly unlucky, it won't be an infinite loop.

Reply:
Still, if it’s prone to cause an infinite loop you should change it

Reply:
There are millions of battles a day on main, and a single infinite loop will freeze the battle process. Based on the last SSB loop, it takes about 2-6 hours to get the loop on average. Never argue the inclusion of a bug is OK because its unlikely, especially one that can basically force a server restart.